go/chat/attachments/preprocess.go

```diff
@@ -189,17 +196,30 @@ func PreprocessAsset(ctx context.Context, log utils.DebugLabeler, filename strin
 	p = Preprocess{
 		ContentType: http.DetectContentType(head),
 	}
-	log.Debug(ctx, "preprocessAsset: detected attachment content type %s", p.ContentType)
 	if _, err := src.Seek(0, 0); err != nil {
 		return p, err
 	}
+	// MIME type detection failed us, try using an extension map
+	if p.ContentType == "application/octet-stream" {
+		var ext string
```

Comment:
i think my comment got lost, but `path.Ext` does this.

https://golang.org/pkg/path/#Ext